src/plopp/backends/matplotlib/line.py

```diff
@@ -57,110 +68,77 @@ def __init__(self, canvas: Canvas, data: sc.DataArray, number: int = 0, **kwargs
+            'color': f'C{artist_number}',
         }
 
-        if data["hist"]:
+        if line_data["hist"]:
             self._line = self._ax.step(
-                data['values']['x'],
-                data['values']['y'],
+                line_data['values']['x'],
+                line_data['values']['y'],
                 label=self.label,
                 zorder=10,
-                **{**default_step_style, **kwargs},
+                **{**default_step_style, **line_args},
             )[0]
 
-            self._mask = self._ax.step(data['mask']['x'], data['mask']['y'])[0]
+            self._mask = self._ax.step(line_data['mask']['x'], line_data['mask']['y'])[
+                0
+            ]
             self._mask.update_from(self._line)
             self._mask.set_color(mask_color)
             self._mask.set_label(None)
             self._mask.set_linewidth(self._mask.get_linewidth() * 3)
             self._mask.set_zorder(self._mask.get_zorder() - 1)
-            self._mask.set_visible(data['mask']['visible'])
+            self._mask.set_visible(line_data['mask']['visible'])
         else:
             self._line = self._ax.plot(
-                data['values']['x'],
-                data['values']['y'],
+                line_data['values']['x'],
+                line_data['values']['y'],
                 label=self.label,
                 zorder=10,
-                **{**default_plot_style, **kwargs},
+                **{**default_plot_style, **line_args},
             )[0]
             self._mask = self._ax.plot(
-                data['mask']['x'],
-                data['mask']['y'],
+                line_data['mask']['x'],
+                line_data['mask']['y'],
                 zorder=11,
                 mec=mask_color,
                 mfc="None",
                 mew=3.0,
                 linestyle="none",
                 marker=self._line.get_marker(),
-                visible=data['mask']['visible'],
+                visible=line_data['mask']['visible'],
             )[0]
 
-        line_mask = ~np.isnan(data['mask']['y'])
-        self._line._plopp_mask = line_mask
+        self.line_mask = sc.array(dims=['x'], values=~np.isnan(line_data['mask']['y']))
 
         # Add error bars
-        if errorbars and (data['stddevs'] is not None):
+        if errorbars and (line_data['stddevs'] is not None):
             self._error = self._ax.errorbar(
-                data['stddevs']['x'],
-                data['stddevs']['y'],
-                yerr=data['stddevs']['e'],
+                line_data['stddevs']['x'],
+                line_data['stddevs']['y'],
+                yerr=line_data['stddevs']['e'],
                 color=self._line.get_color(),
                 zorder=10,
                 fmt="none",
             )
-            # Set the selection mask on the line collection that makes the segments
-            self._error[2][0]._plopp_mask = line_mask[1:] if data["hist"] else line_mask
```

Comment:
Why has this been removed?

Reply:
Good question. This is related to moving the autoscaling from the `Canvas` to the `GraphicalView`.

Before:
The canvas would identify all the graphics it had (looking at matplotlib artists, not our objects) and try to compute the global min/max limits for all artists. Because we wanted to ignore the masked points when searching for the limits, I had to add the mask on the matplotlib object so it could be read at the canvas level.

Now:
The GraphicalView just loops over its Plopp artists, and asks each one of them for their bounding box.
It computes a global bounding box from all of them.
The `Line` artists knows its own mask, so it just returns the corretc bounding box, the View never needs to know about this.